src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -740,6 +742,35 @@ private BoundExpression CreateMethodGroupConversion(SyntaxNode syntax, BoundExpr
             return new BoundConversion(syntax, group, conversion, @checked: false, explicitCastInCode: isCast, conversionGroup, constantValueOpt: ConstantValue.NotAvailable, type: destination, hasErrors: hasErrors) { WasCompilerGenerated = group.WasCompilerGenerated };
         }
 
+        private static bool CheckValidScopedMethodConversion(SyntaxNode syntax, MethodSymbol lambdaOrMethod, TypeSymbol targetType, bool invokedAsExtensionMethod, BindingDiagnosticBag diagnostics)
+        {
+            MethodSymbol? delegateMethod;
+            if (targetType.GetDelegateType() is { } delegateType)
+            {
+                delegateMethod = delegateType.DelegateInvokeMethod;
+            }
+            else if (targetType is FunctionPointerTypeSymbol functionPointerType)
+            {
+                delegateMethod = functionPointerType.Signature;
```

Comment:
>delegateMethod = functionPointerType.Signature [](http://example.com/codeflow?start=16&length=46)

Is wording of ERR_ScopedMismatchInParameterOfTargetDelegate going to be confusing for a function pointer scenario?